Doc 11827 mobile vector search

ElliotFrancisHunter opened the pull request:
Preview: https://preview.docs-test.couchbase.com/Elvec/couchbase-lite/3.2/cbl-whatsnew.html

Beta 1 version of Vector Search in Couchbase Lite

ElliotFrancisHunter committed:
DOC-11827: Initial Vector Search doc commit

ElliotFrancisHunter committed:
DOC-11827: Clear what's new page ready for 3.2 changes

ElliotFrancisHunter committed:
DOC-11827: v1 of vector search page complete

ElliotFrancisHunter committed:
DOC-11827: Updating to correct base

ElliotFrancisHunter committed:
DOC-11827: v1 of the api reference page

ElliotFrancisHunter committed:
DOC-11827: Copying over for objc and android

ElliotFrancisHunter committed:
DOC-11827: Dummy install for swift to be filled.

rajagp commented on modules/ROOT/pages/cbl-whatsnew.adoc:
```diff
@@ -18,19 +18,23 @@ If you are upgrading from 2.x, please refer to the appropriate upgrade page -- s
 
 
-== Release 3.1.0 (April 2023)
+== Release 3.2.0 (March 2024)
```
```suggestion
== Release 3.2.0 - Beta1 (March 2024)
```

ElliotFrancisHunter replied:
Done.

rajagp commented on modules/ROOT/pages/cbl-whatsnew.adoc:
```diff
@@ -18,19 +18,23 @@ If you are upgrading from 2.x, please refer to the appropriate upgrade page -- s
+=== Vector Search
 
-Independent lifecycle management and fine-grained security control simplifies the deployment of multi-tenant and high-application density cloud-to-edge applications at scale.
+AI, LLM, Similarity Search, it's really good trust me.
```
Seems a bit too casual for a developer docs. Also, we are not supporting any AI, LLM stuff. At high level, it should be  "Vector similar search support to power your Generative AI applications. Create vector indexes and use similarity search queries to search for semantically similar items"
 Can you please work with Iveta on a brief description of what to include in "whats new" page

ElliotFrancisHunter replied:
It was placeholder text so I knew to come back to it later. Actual copy is in now.

rajagp commented on modules/ROOT/pages/cbl-whatsnew.adoc:
```diff
@@ -18,19 +18,23 @@ If you are upgrading from 2.x, please refer to the appropriate upgrade page -- s
-// end::scopes-and-collections[]
+=== Extended Date/Time Functionality
+
+Six new functions
```
Please be specific on the overall purpose of these functions- "6 new functions" does not mean much to a developer . Work w/ Iveta on that

rajagp commented on modules/android/nav-android.adoc:
```diff
@@ -28,7 +28,12 @@ ifdef::releasetag[:tag: -{releasetag}]
+  * Search
+    ** xref:android:fts.adoc[Full Text Search]
+    ** xref:android:vector-search.adoc[Vector Search]
+      *** xref:android:vector-search-api-reference.adoc[Vector Search API Reference]
```
This is not "Vector Search API Reference" . Rename to "Index and Query". API references refer to the API definitions that are generated from code - example of API reference is https://docs.couchbase.com/mobile/3.1.6/couchbase-lite-swift/index.html

rajagp commented on modules/android/nav-android.adoc:
```diff
@@ -28,7 +28,12 @@ ifdef::releasetag[:tag: -{releasetag}]
+    ** xref:android:fts.adoc[Full Text Search]
+    ** xref:android:vector-search.adoc[Vector Search]
+      *** xref:android:vector-search-api-reference.adoc[Vector Search API Reference]
+      *** xref:android:working-with-vector-search.adoc[Working with Vector Search]
```
What is this page supposed to cover?

sarahlwelton replied:
I'd recommend switching the order of these two items.

rajagp commented on modules/android/pages/vector-search-api-reference.adoc:
```diff
@@ -0,0 +1,286 @@
+Vector encoding is the process of representing data points such as images, words, or documents as vectors in high-dimensional space where each dimension represents a different attribute of the data. 
+Encoding is a crucial step in preparing data for similarity or nearest neighbour search operations.
+
+Couchbase Lite support three encoding types:
```
There appears to be some confusion around vector encoding. Please catchup with Iveta on this
What you describe here is how Couchbase Lite vector indexes are encoded. 

At a high level, this is what happens
- Data is translated to vector of floats
- Vector index is created on the vectors. You use encoding to save space when storing indexes.

sarahlwelton replied:
```suggestion
Couchbase Lite supports three encoding types:
```

rajagp commented on modules/android/pages/vector-search-api-reference.adoc:
```diff
@@ -0,0 +1,286 @@
+* xref:swift:working-with-vector-search.adoc[Working with Vector Search]
+
+* xref:swift:fts.adoc[Full Text Search]
+
```
General- we must not duplicate what is in API reference guides here (expectation is that we will publish those when we go Beta). We must only include code snippets describing usage and then link to API reference guide for correct API definitions.

rajagp commented on modules/objc/pages/vector-search-api-reference.adoc:
```diff
@@ -0,0 +1,286 @@
+----
+
+You can use less storage by using the `prediction()` function as the encoded vectors will only be stored in the index. 
+However, the index time will be longer as indexing is occurring at run time.
```
```suggestion
However, the index time will be longer as generating of vector embedding is occurring at run time.
```

ElliotFrancisHunter replied:
Much better, thank you!

rajagp commented on modules/objc/pages/vector-search-api-reference.adoc:
```diff
@@ -0,0 +1,286 @@
+
+=== Generate Vectors with Prediction
+
+You can use two methods to generate vectors in Couchbase Lite:
```
The "two methods" has to be described upfront - This section is about generating vectors/ embedding with Prediction function . 
- Embeddings are in line document
- Embeddings generated dynamically during index creation.

ElliotFrancisHunter replied:
Are you saying it should be moved earlier? I'm not entirely sure what you are asking

rajagp commented on modules/objc/pages/vector-search-api-reference.adoc:
```diff
@@ -0,0 +1,286 @@
+
+IMPORTANT: Similar to the Full Text Search `match()` function, `vector_match()` can only be called alone or at the top level `AND` expression.
+
+=== Generate Vectors with Prediction
```
Generate vector embeddings with Prediction function

ElliotFrancisHunter replied:
Done!

rajagp commented on modules/objc/pages/vector-search-api-reference.adoc:
```diff
@@ -0,0 +1,286 @@
+
+[source, sqlpp]
+----
+
```
Shouldn't we describe these examples in context of API? Or is that covered in different section

rajagp commented on modules/swift/pages/vector-search-api-reference.adoc:
```diff
@@ -0,0 +1,286 @@
+
+[source, sqlpp]
+----
+
```
These are swift specific but none of examples show any swift code samples.

ElliotFrancisHunter replied:
I'm still waiting on these. The api doc only contained C snippets for some reason

rajagp commented on modules/swift/pages/vector-search.adoc:
```diff
@@ -0,0 +1,95 @@
+== About Vector Search
+
+You can use Vector Search to efficiently find similar items or content based on the similarity of their vector representations. 
+This is particularly useful for large datasets and real-time response applications.
```
```suggestion
This is useful for building Generative AI applications.
```

ElliotFrancisHunter replied:
Done!

rajagp commented on modules/swift/pages/vector-search.adoc:
```diff
@@ -0,0 +1,95 @@
+
+== About Vector Search
+
+You can use Vector Search to efficiently find similar items or content based on the similarity of their vector representations. 
```
```suggestion
You can use Vector Search to find top N items similar to a given item based on the similarity of their vector representations. 
```

ElliotFrancisHunter replied:
That sounds much better thank you. Done!

sarahlwelton commented on modules/ROOT/pages/cbl-whatsnew.adoc:
```diff
@@ -18,19 +18,23 @@ If you are upgrading from 2.x, please refer to the appropriate upgrade page -- s
+=== Vector Search
 
-Independent lifecycle management and fine-grained security control simplifies the deployment of multi-tenant and high-application density cloud-to-edge applications at scale.
+AI, LLM, Similarity Search, it's really good trust me.
```
I'm guessing this needs to be updated? Lol. Let me know if you need a link to my doc again that explains some of these concepts

rajagp commented on modules/swift/pages/vector-search.adoc:
```diff
@@ -0,0 +1,95 @@
+
+* Recommendations - Recommending items with related text strings
+
+* Search - Ranking results by similarity to given input strings
```
Not sure where you got the list from.   A lot of these appear to be more appropriate for server side and not something that you'd do locally on a device. Search and recommendations are good use cases.  
Also RAG use case- Application can get similarity results from search on CBL and then those results as context to LLM queries for customizing results. 
Please refer to the PRD for benefits of doing this locally -   data privacy, fast response, offline-first, query cost is cheaper

Please sync up with Iveta on this

rajagp commented on modules/swift/pages/vector-search.adoc:
```diff
@@ -0,0 +1,95 @@
+
+* Search - Ranking results by similarity to given input strings
+
+== About Vector Indexes
```
```suggestion
== Vector Indexes
```

rajagp replied:
If we are discussing indexes in overview, shouldn't we also describe search here. So there are two components to vector search support
Index Creation and Search
We should have a section on "Search" that explains that we do similarity search based on distance metric - explain the two that we support. Also mention that we do K- nearest neighbor search.

rajagp replied:
I think we need a high level block diagram at top of page that summarizes how all this comes together
- Document Content -> Generate vector on Content  -> Create vector Index from vectors
- Search query -> Generate vector on query string ("target vector")   -> Vector search on the target vector 
CBL returns top N similar results

rajagp commented on modules/swift/pages/vector-search.adoc:
```diff
@@ -0,0 +1,95 @@
+== About Vector Indexes
+
+Vector Indexes are objects used to efficiently perform similarity search on collections of vectors.
+Before use, a Vector Index needs to be trained to compute the centroids and parameters for encoding the vectors. 
```
Thats not true.  Please watch Jen's show and tell video. Training starts with first query but you need a min number of vectors to train. Prior to that, all vectors are in same partition.

rajagp commented on modules/swift/pages/vector-search.adoc:
```diff
@@ -0,0 +1,95 @@
+
+== About Vector Indexes
+
+Vector Indexes are objects used to efficiently perform similarity search on collections of vectors.
```
```suggestion
Vector Indexes are used to store and manage vector representations of content. They allow for efficient retrieval of  vectors similar to a target vector 
```

ElliotFrancisHunter replied:
Done!

sarahlwelton commented on modules/android/pages/vector-search-api-reference.adoc:
```diff
@@ -0,0 +1,286 @@
+ifdef::show_edition[:page-edition: {release}]
+ifdef::prerelease[:page-status: {prerelease}]
+:page-role:
+:description: API reference for Vector Search in Couchbase Lite.
```
Don't forget to expand this and include it in the body of the page with: 

[abstract]
{description}

sarahlwelton commented on modules/android/pages/vector-search-api-reference.adoc:
```diff
@@ -0,0 +1,286 @@
+== Vector Index API
+
+You can use the Vector Index API's core functionality to efficiently perform similarity search at scale on Vector Indexes.
+You can make use of new qurry functions such as `vector_match()` to query indexed vectors.
```
```suggestion
You can make use of new query functions such as `vector_match()` to query indexed vectors.
```

ElliotFrancisHunter replied:
Done!

sarahlwelton commented on modules/android/pages/vector-search-api-reference.adoc:
```diff
@@ -0,0 +1,286 @@
+
+== Vector Index API
+
+You can use the Vector Index API's core functionality to efficiently perform similarity search at scale on Vector Indexes.
```
```suggestion
You can use the Vector Index API's core functionality to efficiently perform similarity search at scale on Vector Search indexes.
```

ElliotFrancisHunter replied:
Done!

sarahlwelton commented on modules/android/pages/vector-search-api-reference.adoc:
```diff
@@ -0,0 +1,286 @@
+=== Vector Encoding
+
+Vector encoding is the process of representing data points such as images, words, or documents as vectors in high-dimensional space where each dimension represents a different attribute of the data. 
+Encoding is a crucial step in preparing data for similarity or nearest neighbour search operations.
```
```suggestion
Encoding is a crucial step in preparing data for similarity or nearest-neighbor search operations.
```

Watch the British spelling.

pasin replied:
Not sure if `Vector Encoding` mentioned in this section is about the vector encoding methods used in the vector index or not. If it is, Vector Encoding is the method or algorithm that compresses the vectors stored in the vector index. The encoding makes the index size smaller and reduce the I/P during index and query time, however, on the flip side, encoding could make the distance computation and query result less inaccurate.

We support 3 types of encoding in our API: None (No encoding, No data-loss ), SQ and PQ. Here are a bit of info about the SQ and PQ encoding:

https://github.com/couchbaselabs/mobile-vector-search/blob/main/docs/Tuning.md#2-vector-encoding

sarahlwelton commented on modules/android/pages/vector-search-api-reference.adoc:
```diff
@@ -0,0 +1,286 @@
+
+|vectorIndexIdentifier
+|Required
+|The name of the vector index to perform the vector search on.
```
```suggestion
|The name of the vector index to search.
```

sarahlwelton commented on modules/android/pages/vector-search-api-reference.adoc:
```diff
@@ -0,0 +1,286 @@
+
+|===
+
+This function performs vector search against a specific vector index identifier for the specified vector expression.
```
```suggestion
This function performs a vector search against a specific vector index identifier for the specified vector expression.
```
Do we want to be capitalizing Vector Search at all in this description? Try to keep consistent.

sarahlwelton commented on modules/android/pages/vector-search-api-reference.adoc:
```diff
@@ -0,0 +1,286 @@
+
+This function performs vector search against a specific vector index identifier for the specified vector expression.
+If the specified index does not exist, an error will occur on creation of the query.
+The matched vectors will be returned up to the specified limit number, if the limit is not specified then the default value will be used. 
```
```suggestion
The matched vectors will be returned up to the specified limit number. 
If you do not specify a limit, then Couchbase Lite uses a default value of `3`. 
```

sarahlwelton commented on modules/android/pages/vector-search-api-reference.adoc:
```diff
@@ -0,0 +1,286 @@
+The matched vectors will be returned up to the specified limit number, if the limit is not specified then the default value will be used. 
+The returned vectors are sorted by their distance values in ascending order by default.
+
+NOTE: The default value for the `limit` parameter is 3.
```
```suggestion

```

sarahlwelton commented on modules/android/pages/vector-search-api-reference.adoc:
```diff
@@ -0,0 +1,286 @@
+----
+
+=== Vector Distance
+
```
Are you going to add any preamble here?

sarahlwelton replied:
You should try to avoid stacking headings with no content in between.

sarahlwelton commented on modules/android/pages/vector-search-api-reference.adoc:
```diff
@@ -0,0 +1,286 @@
+
+|===
+
+This function returns the distance between the target vector specified in the `vector_match()` function and the matched vector in the specified vector index based on the distance metric set in the index configuration .
```
```suggestion
This function returns the distance between the target vector specified in the `vector_match()` function and the matched vector in the specified vector index, based on the distance metric set in the index configuration.
```

sarahlwelton commented on modules/android/pages/vector-search-api-reference.adoc:
```diff
@@ -0,0 +1,286 @@
+
+You can use two methods to generate vectors in Couchbase Lite:
+
+. You can call a Machine Learning(ML) model, and embed the generated vectors inside the documents.
```
```suggestion
. You can call a Machine Learning (ML) model, and embed the generated vectors inside the documents.
```

sarahlwelton commented on modules/android/pages/vector-search-api-reference.adoc:
```diff
@@ -0,0 +1,286 @@
+
+. You can call a Machine Learning(ML) model, and embed the generated vectors inside the documents.
+
+. You can use the `prediction()` function to generate vectors to be indexed for each document at the indexing time.
```
```suggestion
. You can use the `prediction()` function to generate vectors to be indexed for each document at indexing time.
```

sarahlwelton commented on modules/android/pages/vector-search-api-reference.adoc:
```diff
@@ -0,0 +1,286 @@
+[IMPORTANT]
+--
+
+In the beta release. Updating the index for the documents whose returned vectors from the `prediction()` were missing on calling the ML model is not possible unless you perform the following actions:
```
Unclear on what you're trying to say here?

sarahlwelton commented on modules/android/pages/vector-search.adoc:
```diff
@@ -0,0 +1,95 @@
+ifdef::show_edition[:page-edition: {release}]
+ifdef::prerelease[:page-status: {prerelease}]
+:page-role:
+:description: Use Vector Search for AI applications.
```
Same note about actually using the {description} in your document.

ElliotFrancisHunter replied:
Done!

sarahlwelton commented on modules/android/pages/vector-search.adoc:
```diff
@@ -0,0 +1,95 @@
+You can use Vector Search to efficiently find similar items or content based on the similarity of their vector representations. 
+This is particularly useful for large datasets and real-time response applications.
+
+Vector Search is a Similarity Search method that looks for approximated nearest vectors in localised vector space. 
```
```suggestion
Vector Search is a Similarity Search method that looks for approximated nearest vectors in localized vector space. 
```

ElliotFrancisHunter replied:
Done!

sarahlwelton commented on modules/android/pages/vector-search.adoc:
```diff
@@ -0,0 +1,95 @@
+
+Vector Indexes are objects used to efficiently perform similarity search on collections of vectors.
+Before use, a Vector Index needs to be trained to compute the centroids and parameters for encoding the vectors. 
+Training is configured to commence automatically on the first Vector Search query when there are enough vectors to be trained based on the `minimum-training-size` configuration.
```
```suggestion
Training starts automatically on the first Vector Search query when there are enough vectors to be trained based on the `minimum-training-size` configuration.
```

sarahlwelton commented on modules/android/pages/vector-search.adoc:
```diff
@@ -0,0 +1,95 @@
+Before use, a Vector Index needs to be trained to compute the centroids and parameters for encoding the vectors. 
+Training is configured to commence automatically on the first Vector Search query when there are enough vectors to be trained based on the `minimum-training-size` configuration.
+
+=== Vector Index Configuration
```
Do you have more information to provide context on this table?

Honestly I wouldn't be including content like this in a conceptual topic. This seems out of place.

sarahlwelton commented on modules/android/pages/vector-search.adoc:
```diff
@@ -0,0 +1,95 @@
+|Expression
+|Required
+|No default
+| A SQL++ expression indicating where to get the vectors. 
+A document property for embedded vectors
+`prediction()` to call a registered Predictive model.
```
This still seems like rough notes?

sarahlwelton commented on modules/android/pages/vector-search.adoc:
```diff
@@ -0,0 +1,95 @@
+|Number of Dimensions
+|Required
+|No default
+|2-2048
```
Please provide more context. These numbers aren't useful.

sarahlwelton commented on modules/android/pages/vector-search.adoc:
```diff
@@ -0,0 +1,95 @@
+|Number of Centroids
+|Required
+|No default
+|1-64000. The general guideline is an approximate square root of the number of documents
```
Maybe you should explain what centroids are?

sarahlwelton commented on modules/android/pages/vector-search.adoc:
```diff
@@ -0,0 +1,95 @@
+|Distance Metric
+|Optional
+|Euclidean Distance
+|You can also configure Cosine Distance as your Distance Metric
```
Any information to provide on what a Distance Metric is?

sarahlwelton commented on modules/android/pages/vector-search.adoc:
```diff
@@ -0,0 +1,95 @@
+|Encoding
+|Optional
+| Scalar Quantizer(SQ) or SQ-8 bits
+a|There are three possible configurations:
```
More information on what a quantizer/encoding does would be helpful

sarahlwelton commented on modules/android/pages/vector-search.adoc:
```diff
@@ -0,0 +1,95 @@
+
+|Training Size
+|Optional
+a|There are defaults for both the minimum and maximum training size
```
Okay but what does the training size parameter actually do?

sarahlwelton reviewed (dismissed):
My comments would apply to all of the versions of these files. Try to think more about making the benefits/uses clearer to your reader!

ElliotFrancisHunter committed:
DOC-11827: Adding beta label

ElliotFrancisHunter committed:
DOC-11827: v1 vector search what's new page + deprecation notice

ElliotFrancisHunter committed:
DOC-11827: V1 install for swift, missing specifics

pasin commented on modules/android/pages/vector-search-api-reference.adoc:
```diff
@@ -0,0 +1,287 @@
+
+* No encoding - 4 bytes per dimension, large size with no data loss
+
+* Scalar Quantizer - The default setting.
```
The default encoding is Scalar Quantizer (8 bits).

ElliotFrancisHunter replied:
Done, thank you!

ElliotFrancisHunter committed:
DOC-11827: Add enterprise edition label

ElliotFrancisHunter committed:
DOC-11827: Android prereqs and swift code examples

ElliotFrancisHunter committed:
DOC-11827: Fixes to links, versions numbers and wording feedback

ElliotFrancisHunter committed:
DOC-11827: first round of feedback, language tidying, using descriptions properly etc

ElliotFrancisHunter committed:
DOC-11827: Android install instructions v1

ElliotFrancisHunter committed:
Merge branch 'release/3.1' into DOC-11827-mobile-vector-search

ElliotFrancisHunter committed:
DOC-11827: iOS download instructions v1

ElliotFrancisHunter committed:
DOC-11827: Address feedback for what's new

ElliotFrancisHunter committed:
DOC-11827: typo fix

ElliotFrancisHunter committed:
DOC-11827: platform changes for 3.2

ElliotFrancisHunter committed:
DOC-11827: v2 feedback

ElliotFrancisHunter committed:
DOC-11827: TODO swift snippets

For preview to clarify examples needed.

ElliotFrancisHunter committed:
DOC-11827: Update swift code snippet instructions, update vector search concepts

ElliotFrancisHunter committed:
DOC-11827: v1 snippets copy

ElliotFrancisHunter committed:
DOC-11827: Update snippets

Swift and v1 objc snippets page
Also tidying snippets, correcting links etc

ElliotFrancisHunter committed:
Merge branch 'release/3.2' into DOC-11827-mobile-vector-search

ElliotFrancisHunter committed:
DOC-11827: Android snippets

Start the tab structure for java/kotlin examples
Fixes to typos in comments and extra ::: calling fall over with the tags
Still need to fix this for the java snippets and then tidy up

ElliotFrancisHunter committed:
v1 android snippets copy

Fixed colon fallover issue in snippets
Altered copy of snippets to reflect android snippet code

ElliotFrancisHunter committed:
DOC-11827: Remove dead page

Removing api reference page as unneeded. also removed from the nav

ElliotFrancisHunter committed:
DOC-11827: Install fixes

Removed mention of carthage support and declared lack of support in iOS for Beta.
Updated with the 'beta.1' tag where it was missing.
Explicitly stated 64-bit architecture support only

ElliotFrancisHunter committed:
DOC-11827: Vector Search overview page feedback update.

Update all copy from most recent feedback. Add next steps and update xrefs to remove dead link

ElliotFrancisHunter committed:
DOC-11827: page fixes

Added abstract in, updated and corrected xrefs

ElliotFrancisHunter committed:
DOC-11827: 3.2 Beta fixes and additions

Added beta.1 tag to partials, added beta column to platform support, updated deprecation to 3.2 beta, added DateTime preamble to what's new page

ElliotFrancisHunter committed:
DOC-11827: order fix

osfameron commented on modules/objc/pages/gs-install.adoc:
```diff
@@ -51,66 +57,98 @@ Direct Download::
 
-. Install the pods and open the .xcworkspace file generated by Cocoapods.
++
+<.> Specify the required version number, here we use the latest version of the beta
```
Not urgent, but we're trying to avoid Callouts (see the style guide) these days, so best not to introduce any new ones.

osfameron commented on modules/objc/pages/vector-search.adoc:
```diff
@@ -0,0 +1,137 @@
+
+=== About Dimensions
+
+Vector dimensions describes the amount of numbers in a given vector embedding, commonly known as its' width.
```
no apostrophe after its

osfameron commented on modules/objc/pages/working-with-vector-search.adoc:
```diff
@@ -0,0 +1,319 @@
+
+You can use two methods to generate vectors in Couchbase Lite:
+
+. You can call a Machine Learning(ML) model, and embed the generated vectors inside the documents.
```
still missing a space before parenthesis

osfameron commented on modules/swift/pages/_partials/gs-install-tab-cocoapods.adoc:
```diff
@@ -10,7 +10,7 @@ NOTE: The minimum required Cocoapods  version is *1.9*
 target 'Example' do
   use_frameworks!
-  pod 'CouchbaseLite-Swift', '{version-maintenance}' // <.>
+  pod 'CouchbaseLite-Swift', '{version-maintenance}-beta.1' // <.>
```
OK... we may need to tweak this once we've spoken to build team

osfameron commented on modules/swift/pages/working-with-vector-search.adoc:
```diff
@@ -0,0 +1,310 @@
+
+* The data stored within this index will be of type `string`.
+
+* The width or `dimensions` of the vector index is set to `300`.
```
Absolutely non-urgent, but not sure if you need to put numbers in backticks? 🤔

osfameron commented on modules/swift/pages/working-with-vector-search.adoc:
```diff
@@ -0,0 +1,310 @@
+IMPORTANT: The beta release currently only supports the format of vectors as an array of numbers.
+
+.Vector Index Configuration Options
+[cols ="4*"]
```
weird space

osfameron replied:
More importantly, Put a newline between each ROW (e.g. in this case every 4 pipe character cells.)

osfameron commented on modules/swift/pages/working-with-vector-search.adoc:
```diff
@@ -0,0 +1,310 @@
+|Expression
+|Required
+|No default
+| A SQL++ expression indicating where to get the vectors. 
```
Check that this renders OK. You may need to use `{sqlpp}` to prevent the + signs from getting mangled by asiidoctor

osfameron commented on modules/swift/pages/working-with-vector-search.adoc:
```diff
@@ -0,0 +1,310 @@
+|vectorIndexIdentifier
+|Required
+|The name of the vector index to perform the vector search on.
+|targetVectorExpr
```
newline before new row

osfameron commented on modules/swift/pages/working-with-vector-search.adoc:
```diff
@@ -0,0 +1,310 @@
+
+=== `vector_distance(vectorIndexIdentifier)`
+
+[cols = "3*"]
```
different weird spacing

osfameron reviewed (approved):
mostly niggles, looks good in general

ElliotFrancisHunter commented:
@osfameron will keep this tab open to fix niggly issues after release
